next-app/src/routes/password/change/+page.svelte

```diff
@@ -1,39 +1,36 @@
-<script>
-	import { goto } from '$app/navigation'
-	import { UPDATE } from '$lib/fetch'
+<script lang=ts>
 	import {
 		Button,
 		Form,
 		Input,
 	} from '$lib/forms'
 	import PageHeader from '$lib/PageHeader.svelte'
+	import { error } from '$lib/error'
+    import type { ActionData } from './$types'
+
+	export let form: ActionData
 
 	let new_password = ''
 	let new_password_confirm = ''
 
-	async function change_password() {
-		await UPDATE({url: '/password', body: {
-			password: new_password,
-			password_confirm: new_password_confirm
-		}})
-
-		goto('/password/changed')
+	$: if (form?.failed) {
+		$error = Error(form.failed)
 	}
 </script>
 
 <svelte:head>
 	<title>Change your password</title>
 </svelte:head>
 
-<div class='mx-auto max-w-lg'>
+<section class='mx-auto max-w-lg'>
 	<PageHeader>
 		Change your password
 	</PageHeader>
 
-	<Form on:submit={ change_password }>
-		<Input label='New password:' type=password bind:value={ new_password } required autofocus />
-		<Input label='Confirm new password:' type=password bind:value={ new_password_confirm } required />
+	<Form>
```

Comment:
I know that by convention since this is under the route 'password/change' that it'll get handled on the corresponding `page.server.ts` but I would like to propose a convention to always specify the action attribute (in this case it should be `/password/change` if I understand correctly) so that if this code gets refactored to another page (potentially via a component) it'll still work.

Reply:
I'm a bit torn on this, because I think one of the selling points of Svelte is that so much is implicit in the architecture that there is simply less code to write.  I can really go either way.  It's good to have conventions especially since we are a team working on shared code.  It's also looks elegant to see a simple `<form>` element that just works.

Reply:
> to always specify the action attribute

I like that kind of defensive coding thought.  In my mind these are so tightly coupled, any refactoring, especially of a route, is going to come with changes to both.  I'm open to addressing this if/when a situation comes up and thinking through convention again at that point.

Reply:
alright that sounds good. To me it was less refactoring the route, and more moving the form code into a shared component that might be used on another page.

Reply:
at that point, we might be more inclined to move toward an "endpoint" rather than an "action"